energy-integration/governance-v2/src/views.rs

```diff
@@ -41,27 +38,59 @@ pub trait ViewsModule:
             return GovernanceProposalStatus::Active;
         }
 
-        if self.quorum_and_vote_reached(proposal_id) {
+        if self.quorum_reached(proposal_id) && self.vote_reached(proposal_id) {
             GovernanceProposalStatus::Succeeded
+        } else if self.vote_down_with_veto(proposal_id) {
+            GovernanceProposalStatus::DefeatedWithVeto
         } else {
             GovernanceProposalStatus::Defeated
         }
     }
 
-    fn quorum_and_vote_reached(&self, proposal_id: ProposalId) -> bool {
+    fn vote_reached(&self, proposal_id: ProposalId) -> bool {
         let proposal_votes = self.proposal_votes(proposal_id).get();
         let total_votes = proposal_votes.get_total_votes();
         let total_up_votes = proposal_votes.up_votes;
-        let total_down_votes = proposal_votes.down_votes;
         let total_down_veto_votes = proposal_votes.down_veto_votes;
         let third_total_votes = &total_votes / 3u64;
-        let quorum = self.quorum().get();
+        let half_total_votes = &total_votes / 2u64;
 
         if total_down_veto_votes > third_total_votes {
             false
         } else {
-            total_votes >= quorum && total_up_votes > (total_down_votes + total_down_veto_votes)
+            total_up_votes > half_total_votes
+        }
+    }
+
+    fn vote_down_with_veto(&self, proposal_id: ProposalId) -> bool {
+        let proposal_votes = self.proposal_votes(proposal_id).get();
+        let total_votes = proposal_votes.get_total_votes();
+        let total_down_veto_votes = proposal_votes.down_veto_votes;
+        let third_total_votes = &total_votes / 3u64;
+
+        total_down_veto_votes > third_total_votes
+    }
+
+    fn quorum_reached(&self, proposal_id: ProposalId) -> bool {
+        let proposal = self.proposals().get(proposal_id);
+        let total_energy_for_proposal = proposal.total_energy;
+
+        let required_minimum_percentage = proposal.minimum_quorum;
+
+        let current_quorum = self.get_current_quorum(proposal_id);
+        let current_quorum_percentage =
+            current_quorum * FULL_PERCENTAGE / total_energy_for_proposal;
+
+        current_quorum_percentage > required_minimum_percentage
```

Comment:
Not very important, but for clarity, maybe check if `current_quorum_percentage` is greater or equal than `required_minimum_percentage`?